### ISSUE TITLE
Use a threadpool for static items

### ISSUE THREAD

**Issue by sector-f** [sector-f/admiral#4]
Currently a new thread is spawned for every occurrence of every item specified in `items`. As thread spawning is somewhat expensive, this is not preferable. It would probably be beneficial to use a [threadpool](https://crates.io/crates/threadpool) (perhaps with only a single worker) for items marked as `static` so as to reduce the number of threads spawned.

**Comment by Henning-K** [comment]
I'd like to work on this one.
I assume the part you'd like changed is in main.rs L132-L134?

**Comment by Henning-K** [comment]
I assume threads should be spawned on the ThreadPool when they contain a `true` static key and through the usual `thread::spawn` method if they don't contain the "static" key or that key is `false`?

**Comment by sector-f** [comment]
> I assume the part you'd like changed is in main.rs L132-L134?

Correct. You'll also want to remove the check [here](https://github.com/sector-f/admiral/blob/master/src/main.rs#L29-L31).

> I assume threads should be spawned on the ThreadPool when they contain a true static key and through the usual thread::spawn method if they don't contain the "static" key or that key is false?

That does sound right, but you don't need to check the keys yourself. They're already checked when the `Command` is created, so just use the `is_static` field.

So, [this](https://github.com/sector-f/admiral/blob/master/src/main.rs#L130-L134) should look something like:

````rust
let command = command::get_command(&value, &table, position);

if command.is_static {
    // Threadpool stuff
} else {
    let _ = thread::spawn(move || {
        run_command(command, clone);
    });
}
````

**Comment by Henning-K** [comment]
Made the changes, will open the PR next week (when it's October).

**Comment by sector-f** [comment]
Fixed by #5
